### ISSUE TITLE
Support lazy.nvim configuration

### ISSUE THREAD

**Issue by sjclayton** [AntonVanAssche/music-controls.nvim#1]
**Describe the solution you'd like**

It would be nice if there was native support for configuring this plugin using the default plugin spec of [lazy.nvim](https://github.com/folke/lazy.nvim) as it is very popular and lots of people are using it.

It provides for passing options directly to a plugin, rather than polluting the `_G.{variable}` global namespace with extra variables.

**Additional context**

*Examples*

My current lazy config for this plugin

```lua
{
    'AntonVanAssche/music-controls.nvim',
    dependencies = { 'rcarriga/nvim-notify' },
    keys = {
      { '<localleader>j', '<CMD>MusicPlay<CR>', desc = 'Music - play/pause' },
      { '<localleader>k', '<CMD>MusicNext<CR>', desc = 'Music - next track' },
      { '<localleader>l', '<CMD>MusicPrev<CR>', desc = 'Music - prev track' },
      { '<localleader>;', '<CMD>MusicCurrent<CR>', desc = 'Music - current track' },
    },
    config = function()
      _G.music_controls_default_player = 'spotify'
    end,
}
```

What it would look like using lazy spec

```lua
{
    'AntonVanAssche/music-controls.nvim',
    dependencies = { 'rcarriga/nvim-notify' },
    keys = {
       -- user keymaps here
    },
    opts = {
       default_player = 'spotify'
    },
}
```

Thanks!!

**Comment by AntonVanAssche** [comment]
Hi @sjclayton,

I've added your feature request in the `opts-setup` branch (see commit: 457d91a). Feel free to test it before I merge it into the `master` branch.

The changes introduced in the above commit allow you to configure the plugin in the way you requested.

```lua
return {
  "AntonVanAssche/music-controls.nvim",
  branch = "opts-setup", -- (Temporary for testing)
  opts = {
    default_player = "spotify"
  }
}
```

**Comment by AntonVanAssche** [comment]
Changes related to documentation have also been added to the same commit.

**Comment by sjclayton** [comment]
@AntonVanAssche 

Every thing looks good...  only one thing I noticed was on line 3 in the plugin's `init.lua` that the require call for `notify` should probably be wrapped inside a pcall, so `vim.notify = pcall(require('notify')())` as to prevent the plugin from throwing an error if the user happens to not have nvim-notify installed, if you are going to be removing it as a dependency for your plugin, which your update to the docs seems to indicate.

 I have not had the time yet to test the new implementation fully however, as I am in the middle of doing a major rewrite on my config.

Thanks for implementing this!

**Comment by AntonVanAssche** [comment]
> Every thing looks good... only one thing I noticed was on line 3 in the plugin's `init.lua` that the require call for `notify` should probably be wrapped inside a pcall, so `vim.notify = pcall(require('notify')())` as to prevent the plugin from throwing an error if the user happens to not have nvim-notify installed, if you are going to be removing it as a dependency for your plugin, which your update to the docs seems to indicate.

This was not the intention, I modified the commit to include the dependency again. 😅

Thanks for noticing!

**Comment by sjclayton** [comment]
Tested, it's all good to merge.

**Comment by dougreese** [comment]
I am not using lazy.vim (vim-plug user), but also wanted to set the player to 'spotify' on startup. I found that by adding the following to my existing lua setup configuration in .vimrc works.

```lua
require("music-controls").setup({
  default_player = 'spotify',
})
```

If you don't currently have a lua setup section in .vimrc, it might look something like:

```lua
lua << END
require("music-controls").setup({
  default_player = 'spotify',
})
END
```

I figured this might help someone else. Thanks for the handy plugin.
